usaspending_api/etl/elasticsearch_loader_helpers/controller.py

```diff
@@ -93,31 +96,34 @@ def complete_process(self) -> None:
             )
             update_last_load_date(f"{self.config['stored_date_key']}", self.config["processing_start_datetime"])
 
-    def construct_tasks(self):
+    def determine_partitions(self) -> int:
+        """Create partion size less than or equal to max_size for more even distribution"""
+        if self.config["partition_size"] > self.record_count:
+            return 1
+        return ceil(self.record_count / self.config["partition_size"])
```

Comment:
What about the indexing on modulo 10000? This will yield a variable number of partitions. Does it slow the SQL down? If not will the index be removed?

Reply:
You're giving me mixed signals. I re-added `es_rapidloader` to this PR could be merged. I also switched to parameterizing the desired partition size rather than the number of partitions.

I'm not suggesting the new script is remotely close to the performance of the existing script with the current SQL views. There are no changes to the DB in this PR.

Reply:
I was asking about the sql performance. You had said having the DB index and fixing partitions to 10k was to make the sql perform better. I was okay with that rationale for keeping the `partitions` arg at 10k partitions, and wouldn't want to tank SQL performance if that crutch was necessary. 

As you said though -- given awards/transactions will still use the old way for a bit, maybe this is something we iterate on and try different things.

1. Try accounts data using this ( modulo of id against count/partition_size )
2. If lag on sql query, try to inject ID ranges as the partitioning strategy (what spark happens to do right now that shows good performance -- esp. because the plan is to reindex ALL accounts data each night = less skew)
3. After delta views are pushed down, try out different strategies for that data: 1) modulo, 2) fixed 10k partitions, 3) id ranges

Reply:
Materializing the data in the view is the single-biggest performance improvement.